contracts/Witch.sol

```diff
@@ -9,79 +9,141 @@ import "./math/WMul.sol";
 import "./math/WDiv.sol";
 import "./math/WDivUp.sol";
 import "./math/CastU256U128.sol";
+import "./math/CastU256U32.sol";
 
 
 contract Witch is AccessControl() {
     using WMul for uint256;
     using WDiv for uint256;
     using WDivUp for uint256;
     using CastU256U128 for uint256;
+    using CastU256U32 for uint256;
 
-    event AuctionTimeSet(uint128 indexed auctionTime);
-    event InitialProportionSet(uint128 indexed initialProportion);
+    event DurationSet(uint32 indexed duration);
+    event InitialOfferSet(uint64 indexed initialOffer);
+    event DustSet(uint128 indexed dust);
     event Bought(bytes12 indexed vaultId, address indexed buyer, uint256 ink, uint256 art);
+    event Auctioned(bytes12 indexed vaultId, uint256 indexed start);
   
-    uint128 public auctionTime = 4 * 60 * 60; // Time that auctions take to go to minimal price and stay there.
-    uint128 public initialProportion = 5e17;  // Proportion of collateral that is sold at auction start.
+    struct Auction {
+        address owner;
+        uint32 start;
+    }
+
+    uint32 public duration = 4 * 60 * 60; // Time that auctions take to go to minimal price and stay there.
+    uint64 public initialOffer = 5e17;  // Proportion of collateral that is sold at auction start (1e18 = 100%)
+    uint128 public dust;                     // Minimum collateral that must be left when buying, unless buying all
```

Comment:
This the same for all collateral types? That doesn't seem right.

Reply:
Would only dust have to be different for all collateral types, or also the auction duration and the initial proportion of the collateral that is offered?